isobus/src/isobus_virtual_terminal_client_update_helper.cpp

```diff
@@ -0,0 +1,82 @@
+//================================================================================================
+/// @file isobus_virtual_terminal_client_update_helper.cpp
+///
+/// @brief A helper class to update and track the state of an active working set.
+/// @author Daan Steenbergen
+///
+/// @copyright 2023 The Open-Agriculture Developers
+//================================================================================================
+
+#include "isobus/isobus/isobus_virtual_terminal_client_update_helper.hpp"
+
+#include "isobus/isobus/can_stack_logger.hpp"
+
+namespace isobus
+{
+	VirtualTerminalClientUpdateHelper::VirtualTerminalClientUpdateHelper(std::shared_ptr<VirtualTerminalClient> client) :
+	  VirtualTerminalClientStateTracker(client->get_internal_control_function()),
```

Comment:
Possibility to crash here if the shared_ptr is invalid... would be nice if we could catch that with an assert or something, or maybe a `nullptr == client ? nullptr :  client->get_internal_control_function()`